pkg/kubelet/cm/cpumanager/state/state_file.go

```diff
@@ -0,0 +1,175 @@
+	"io/ioutil"
+	"k8s.io/kubernetes/pkg/kubelet/cm/cpuset"
+	"os"
+	"sync"
+)
+
+type stateFileData struct {
+	DefaultCPUSet string            `json:"defaultCpuSet"`
+	Entries       map[string]string `json:"reservedList,omitempty"`
+}
+
+type stateFile struct {
+	sync.RWMutex
+	stateFilePath string
+	State
+}
+
+// NewFileState creates new State for keeping track of cpu/pod assignment with file backend
+func NewFileState(filePath string) State {
+	stateFile := &stateFile{
+		stateFilePath: filePath,
+		State:         NewMemoryState(),
+	}
+	stateFile.tryRestoreState()
+	return stateFile
+}
+
+// tryRestoreState tries to read state file, upon any error,
+// err message is logged and state is left clean. un-initialized
+func (sf *stateFile) tryRestoreState() {
+	sf.Lock()
+	defer sf.Unlock()
+	var err error
+	invalidStateFile := false
+
+	// used when all parsing is ok
+	tmpAssignments := make(ContainerCPUAssignments)
+	tmpDefaultCPUSet := cpuset.NewCPUSet()
+	tmpContainerCPUSet := cpuset.NewCPUSet()
+
+	var content []byte
+
+	if content, err = ioutil.ReadFile(sf.stateFilePath); os.IsNotExist(err) {
+		// Create file
+		if _, err = os.Create(sf.stateFilePath); err != nil {
+			glog.Errorf("[cpumanager] state file: unable to create state file \"%s\":%s", sf.stateFilePath, err.Error())
+			panic("[cpumanager] state file not created")
+		}
+		glog.Infof("[cpumanager] state file: created empty state file \"%s\"", sf.stateFilePath)
+	} else {
+		// File exists - try to read
+		var readState stateFileData
+
+		if err = json.Unmarshal(content, &readState); err != nil {
+			glog.Warningf("[cpumanager] state file: could not unmarshal, corrupted state file - \"%s\"", sf.stateFilePath)
+			invalidStateFile = true
+		}
+
+		if !invalidStateFile {
+			if tmpDefaultCPUSet, err = cpuset.Parse(readState.DefaultCPUSet); err != nil {
+				glog.Warningf("[cpumanager] state file: could not parse state file - [defaultCpuSet:\"%s\"]", readState.DefaultCPUSet)
+				invalidStateFile = true
+			}
+		}
+
+		if !invalidStateFile {
+			for containerID, cpuString := range readState.Entries {
+				if tmpContainerCPUSet, err = cpuset.Parse(cpuString); err != nil {
+					glog.Warningf("[cpumanager] state file: could not parse state file - container id: %s, cpuset: \"%s\"", containerID, cpuString)
+					invalidStateFile = true
+				}
+				tmpAssignments[containerID] = tmpContainerCPUSet
+			}
+		}
+
+		if invalidStateFile {
+			sf.State.ClearState()
+			sf.storeState()
+		} else {
+			sf.State.SetDefaultCPUSet(tmpDefaultCPUSet)
+			sf.State.SetCPUAssignments(tmpAssignments)
+
+			glog.V(2).Infof("[cpumanager] state file: restored state from state file \"%s\"", sf.stateFilePath)
+			glog.V(2).Infof("[cpumanager] state file: defaultCPUSet: %s", tmpDefaultCPUSet.String())
+		}
+	}
+	return
+}
+
+// saves state to a file, caller is responsible for locking
+func (sf *stateFile) storeState() error {
+	var content []byte
+	var err error
+
+	writeState := stateFileData{
```

Comment:
Nit: another (non-verb) name for this would be preferable. How about `data`?

Reply:
:+1: